text/001x-battery_improvements.md

```diff
@@ -0,0 +1,149 @@
+  * Start date: 2022-03-26
+  * Contributors: Hamish Willee <hamishwillee@gmail.com>, ...
+  * Related issues: 
+    - [16-cell support #1747](https://github.com/mavlink/mavlink/pull/1747)
+	- [common: added voltage and current multipliers to BATTERY_STATUS #233](https://github.com/ArduPilot/mavlink/pull/233)
+    - [Add MAV_BATTERY_CHARGE_STATE_CHARGING #1068](https://github.com/mavlink/mavlink/pull/1068)
+
+  
+# Summary
+
+This RFC proposes:
+- a new `BATTERY_STATUS_V2` message that has a single cumulative voltage value rather than individual cell voltage arrays.
+- a new (optional) `BATTERY_VOLTAGES` message that can be scaled to as many cells as needed, and which reports them as faults.
+- mechanisms to ease supporting both message types until we can eventually deprecate `BATTERY_STATUS`
+
+  
+# Motivation
+
+The motivation is to:
+- reduce the memory required for battery reporting ([anecdotal evidence](https://github.com/ArduPilot/mavlink/pull/233#issuecomment-976197179) indicates cases with 15% of bandwidth on a default RF channel).
+- Provide a cleaner message and design for both implementers and consumers.
+
+The [BATTERY_STATUS](https://mavlink.io/en/messages/common.html#BATTERY_STATUS) has two arrays that can be used to report individual cell voltages (up to 14), or a single cumulative voltage, and which cannot be truncated.
+The vast majority of consumers are only interested in the total battery voltage, and either sum the battery cells or get the cumulative voltage.
+By separating the cell voltage reporting into a separate message the new battery status can be much smaller, and the cell voltages need only be sent if the consumer is actually interested.
+
+# Detailed Design
+
+There are three parts to the design:
+1. A more efficient status message
+2. A scalable battery voltage message.
+3. Mechanisms that allow the new messages to seamlessly coexist with the old one along with eventual deprecation of the older message.
+
+## BATTERY_STATUS_V2
+
+The proposed message is:
+
+```xml
+    <message id="???" name="BATTERY_STATUS_V2">
+      <description>Battery dynamic information. This should be streamed (nominally at 1Hz). Static battery information is sent in SMART_BATTERY_INFO.</description>
+      <field type="uint8_t" name="id" instance="true">Battery ID</field>
```

Comment:
Feedback @WickedShell
- voltage/current/consumed/ are needed at high rate.
- temperature could be low - would not mind
- Actually, mostly I'd be interested in a `POWER_STATUS` message that has just voltage + current and an ID. This would not be battery specific.
- The mode and hot swap enum stuff is odd and should be removed.
- For cell voltages, want to be able to log long term traces of all cells.

Reply:
consumed and percent_remaining are essentially the same thing, do we need both? I agree with needing some way to view cell voltages. I think a separate message for cell voltages that can be enabled would be a good idea.

Reply:
@dakejahl 
1. Yes, separate message for cell voltages defined. Search on `BATTERY_CELL_VOLTAGES` below.
2.  > consumed and percent_remaining are essentially the same thing, do we need both?

   It might be worth having both - I will add this as a "question". Most recipients - e.g. a GCS or FPV will only be interested in displaying the `percent_remaining` (or perhaps "time remaining") but that is pretty low resolution. However you don't want to force very basic systems to have to calculate this.

Reply:
Marking as resolved.